### ISSUE TITLE
NullPointerException after upgrade to 3.12.0

### ISSUE THREAD

**Issue by msymons** [jenkinsci/cucumber-reports-plugin#200]
After upgrade to plugin v3.12.0 from versions including v3.10.3 and v3.11.0, builds are failing with NullPointerException

```
[CucumberReport] Processing 1 json files:
[CucumberReport] /xxx/builds/689/cucumber-html-reports/.cache/cucumber-html-report/cucumber.json
ERROR: Build step failed with exception
java.lang.NullPointerException
	at net.masterthought.jenkins.CucumberReportPublisher.generateReport(CucumberReportPublisher.java:289)
	at net.masterthought.jenkins.CucumberReportPublisher.perform(CucumberReportPublisher.java:229)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:78)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:720)
	at hudson.maven.MavenModuleSetBuild$MavenModuleSetBuildExecution.post2(MavenModuleSetBuild.java:1073)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:665)
	at hudson.model.Run.execute(Run.java:1753)
	at hudson.maven.MavenModuleSetBuild.run(MavenModuleSetBuild.java:543)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:405)
Build step 'Cucumber reports' marked build as failure
Skipping sonar analysis due to bad build status FAILURE
```
It seems that this might be a regression of #182, which was itself a regression of #176.

Examining job config.xml immediately following upgrade, we see:

```
<net.masterthought.jenkins.CucumberReportPublisher plugin="cucumber-reports@3.12.0">
  <fileIncludePattern>**/cucumber.json</fileIncludePattern>
  <fileExcludePattern></fileExcludePattern>
  <jsonReportDirectory></jsonReportDirectory>
  <failedStepsNumber>0</failedStepsNumber>
  <skippedStepsNumber>0</skippedStepsNumber>
  <pendingStepsNumber>0</pendingStepsNumber>
  <undefinedStepsNumber>0</undefinedStepsNumber>
  <failedScenariosNumber>0</failedScenariosNumber>
  <failedFeaturesNumber>0</failedFeaturesNumber>
  <trendsLimit>0</trendsLimit>
  <parallelTesting>false</parallelTesting>
  <classifications class="empty-list"/>
</net.masterthought.jenkins.CucumberReportPublisher>
```
Where happened to sortingMethod?  It was there before upgrade.  Running with the above config will result in the NullPointerException.

If one opens the job configuration screen and then save (not even needing to make a change), then config updates to:

```
<net.masterthought.jenkins.CucumberReportPublisher plugin="cucumber-reports@3.12.0">
  <fileIncludePattern>**/cucumber.json</fileIncludePattern>
  <fileExcludePattern></fileExcludePattern>
  <jsonReportDirectory></jsonReportDirectory>
  <failedStepsNumber>0</failedStepsNumber>
  <skippedStepsNumber>0</skippedStepsNumber>
  <pendingStepsNumber>0</pendingStepsNumber>
  <undefinedStepsNumber>0</undefinedStepsNumber>
  <failedScenariosNumber>0</failedScenariosNumber>
  <failedFeaturesNumber>0</failedFeaturesNumber>
  <trendsLimit>0</trendsLimit>
  <parallelTesting>false</parallelTesting>
  <sortingMethod>ALPHABETICAL</sortingMethod>
  <classifications class="empty-list"/>
  <classificationsFilePattern></classificationsFilePattern>
</net.masterthought.jenkins.CucumberReportPublisher>   
```

With this configuration, the job runs fine and cucumber reports are generated successfully.

**Comment by damianszczepanik** [comment]
I will prepare fix for this. In the meantime as the hot fix you can edit the job configuration and save it (without any changes) so NPE should be replaced as empty collection

**Comment by msymons** [comment]
Thanks for the speedy fix.  

I had one Jenkins server still running an older plugin and so was able to test upgrade from v3.11.0 to the fixed v3.12.1.   No problems!
